### PR TITLE
Various fixes

### LINKED ISSUE #116: user version of the app shows the debug console
version: 928 and latest 

Having the debug console is handy for us developers, but it is very ugly for the final users.

To solve this issue, modify our ci workflow to perform two different types of builds, debug and release. Release should overwrite the project configuration like this ``-p:OutputType=WinExe``

### LINKED ISSUE #129: Add Button To Join Discord
image

Add Button To Join Discord

### LINKED ISSUE #164: Hide tabs for unfinished features
News and Settings tab don't do anything currently and should be hidden for the users.

### PR DESCRIPTION
**Pull request by CorruptComputer**
Changes:
- Hides the debug console (Fixes #116)
- Adds a way to hide WIP Tabs like the Settings tab (Fixes #164)
- Adds a 'By: <>' field to the changelog
- Adds a Flatpak build to the CI so failures there can be caught in the future
- Adds a link to the Discord and the Launcher issues (Fixes #129)
- Includes some general code cleanup around the areas I was editing

image

This has been tested on Linux and in a Windows VM.

### CODE REVIEW

File: UnitystationLauncher/ViewModels/ChangelogViewModel.cs [hunk 1]
```diff
@@ -1,41 +1,52 @@
...
             _client = new GitHubClient(new ProductHeaderValue("UnitystationCommitNews"));
-            RxApp.MainThreadScheduler.ScheduleAsync((scheduler, ct) => GetPullRequestsAsync());
+            PullRequests = new();
+
+            RxApp.TaskpoolScheduler.ScheduleAsync(GetPullRequestsAsync);
         }
 
-        public async Task GetPullRequestsAsync()
+        private async Task GetPullRequestsAsync(IScheduler scheduler, CancellationToken cancellationToken)
         {
```
Comment (CorruptComputer) [inline UnitystationLauncher/ViewModels/ChangelogViewModel.cs:27]: Codacy says to remove this, but this is part of the required method signature by the task scheduler. Even if we aren't using these parameters we have to have them.

Comment (CorruptComputer) [reply UnitystationLauncher/ViewModels/ChangelogViewModel.cs:27]: image

Removing this:
image

File: UnitystationLauncher/ViewModels/ChangelogViewModel.cs [hunk 2]
```diff
@@ -1,41 +1,52 @@
...
             _client = new GitHubClient(new ProductHeaderValue("UnitystationCommitNews"));
-            RxApp.MainThreadScheduler.ScheduleAsync((scheduler, ct) => GetPullRequestsAsync());
+            PullRequests = new();
+
+            RxApp.TaskpoolScheduler.ScheduleAsync(GetPullRequestsAsync);
         }
 
-        public async Task GetPullRequestsAsync()
+        private async Task GetPullRequestsAsync(IScheduler scheduler, CancellationToken cancellationToken)
```
Comment (corp-0) [inline UnitystationLauncher/ViewModels/ChangelogViewModel.cs]:
```suggestion
        private async Task GetPullRequestsAsync(IScheduler _, CancellationToken cancellationToken)
```

this should suffice to comply with the signature and communicate it is an unused parameter

File: UnitystationLauncher/ViewModels/ChangelogViewModel.cs [hunk 3]
```diff
@@ -1,41 +1,52 @@
...
-        readonly GitHubClient _client;
-        ObservableCollection<PullRequestViewModel> _pullRequests = new ObservableCollection<PullRequestViewModel>();
+        private readonly GitHubClient _client;
 
-        public ObservableCollection<PullRequestViewModel> PullRequests
-        {
-            get => _pullRequests;
-            set => this.RaiseAndSetIfChanged(ref _pullRequests, value);
-        }
+        ObservableCollection<PullRequestViewModel> PullRequests { get; }
```
Comment (corp-0) [inline UnitystationLauncher/ViewModels/ChangelogViewModel.cs]:
```suggestion
        private ObservableCollection<PullRequestViewModel> PullRequests { get; }
```

I understand this didn't have it before, but please prefer to use explicit access modifier.

File: UnitystationLauncher/Program.cs [hunk 4]
```diff
@@ -2,21 +2,18 @@
...
             .UsePlatformDetect()
-            .With(new X11PlatformOptions { UseGpu = true })
-            .With(new AvaloniaNativePlatformOptions { UseGpu = true, UseDeferredRendering = false })
-            .With(new MacOSPlatformOptions { ShowInDock = true })
-            .With(new Win32PlatformOptions
-            {
-                UseDeferredRendering = false
-            })
+            .With(new AvaloniaNativePlatformOptions { UseGpu = true, UseDeferredRendering = false }) // Common options
+                                                                                                     //.With(new Win32PlatformOptions {  }) // Windows Specific
```
Comment (corp-0) [inline UnitystationLauncher/Program.cs]: what's up with the commented out code?

**Review by corp-0 — CHANGES_REQUESTED** [review]

File: UnitystationLauncher/Program.cs (hunk 4 above)
Comment (CorruptComputer) [reply UnitystationLauncher/Program.cs]: I had commented these out while testing and forgot to remove them. Will fix that.

**Review by corp-0 — DISMISSED** [review]

File: UnitystationLauncher/Program.cs (hunk 4 above)
Comment (CorruptComputer) [reply UnitystationLauncher/Program.cs]: Should be fixed now, I also updated the comments to explain a bit better. They were slightly misleading before, as those were just my guesses as to what they were. I checked the Avalonia source code for them now, so these should be correct.

File: UnitystationLauncher/ViewModels/ChangelogViewModel.cs (hunk 3 above)
Comment (CorruptComputer) [reply UnitystationLauncher/ViewModels/ChangelogViewModel.cs]: No worries, please do tell me about these if you see them. Getting the code to be more standardized is always good.

File: UnitystationLauncher/ViewModels/ChangelogViewModel.cs (hunk 2 above)
Comment (CorruptComputer) [reply UnitystationLauncher/ViewModels/ChangelogViewModel.cs]: Ah yeah thanks, should be fixed now.

**Comment by corp-0** [comment]
seems like codacy is eating glue, can you push an empty commit to force it run again?

**Comment by corp-0** [comment]
I dismissed that codacy issue, can you trigger ci again, please?

**Review by corp-0 — APPROVED** [review]

**Pull request merged**